### PR TITLE
Add scrollbar

### PR DESCRIPTION
**Pull request by SuperConfuserUser**
Add horizontal and vertical scrollbar UI component. Handles scrolling, dragging, clicking, and key down input.

Future tasks include:
- handling updates on content or container resize
- a better system to filter input or manage the paper tools

### CODE REVIEW

File: src/components/scrollbar.test.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import * as paper from 'paper';
```
Comment (cfsnyder) [inline src/components/scrollbar.test.ts:1]: nice tests @SuperConfuserUser !

File: src/components/scrollbar.ts [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+type ScrollbarDirection = 'horizontal' | 'vertical';
+
+/**
+ * Interface for scrollable content.
+ */
+interface Scrollable {
+  /**
+   * @property content The content that will be scrolled.
+   */
+  content: paper.Group | paper.Item;
```
Comment (cfsnyder) [inline src/components/scrollbar.ts:22]: nice docs!

File: src/components/scrollbar.ts [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+   * });
+   */
+  setCustomEffects(customEffects: CustomEffects): void {
+    this.customEffects = customEffects;
+  }
+
+  /**
+   * Gets the scrollbar element.
+   */
+  getScrollbar(): paper.Path {
```
Comment (cfsnyder) [inline src/components/scrollbar.ts:223]: It's better to be explicit about the possibility that a return value may be undefined. e.g. you could express the return type as a type union using `paper.Path | undefined`. It is also nice to document the conditions under which an undefined value is expected in the tsdoc.

File: src/components/scrollbar.ts [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+  /**
+   * Set a custom scrollbar.
+   * Most attributes can be updated with common paper.Item accessors via getScrollbar(), but something like the radius
+   * of a rounded rectangle is difficult to change. To completely customize the scrollbar, create a new one in the
+   * parent and use setScrollbar(). This allows the scrollbar to retain all of the applied behaviors.
+   *
+   * @param customScrollbar
+   *
+   * @example
+   * const scrollbar = new ScrollbarComponent(...);
```
Comment (cfsnyder) [inline src/components/scrollbar.ts:236]: thanks for providing awesome examples @SuperConfuserUser !

File: src/components/scrollbar.test.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  test('does not build when content fits inside container' , () => {
+    const scrollable = {
+      content: new paper.Path.Rectangle(
+        new paper.Point(0, 0),
+        new paper.Size(500, 500)
+      ),
+      containerBounds: new paper.Rectangle(0, 0, 500, 500),
+    };
+    const scrollbar = new ScrollbarComponent(scrollable, new paper.Point(0, 0));
+    expect(scrollbar.getScrollbar()).toBeFalsy();
```
Comment (cfsnyder) [inline src/components/scrollbar.test.ts]: It is better to be explicit about the expectation vs. using `falsy`/`truthy`. i.e. I'd rather assert that these are *precisely* equal to `undefined` if that is what we really expect.

**Review by cfsnyder — COMMENTED** [review]
As usual, this PR looks excellent @SuperConfuserUser! I just added a few minor comments related to code style. Great job!

File: src/components/scrollbar.ts (hunk 3 above)
Comment (SuperConfuserUser) [reply src/components/scrollbar.ts:223]: fixed. should always return a scrollbar now.

File: src/components/scrollbar.test.ts (hunk 5 above)
Comment (SuperConfuserUser) [reply src/components/scrollbar.test.ts]: fixed. tests are more explicit now.

**Review by cfsnyder — APPROVED** [review]
Awesome, Chely!

**Pull request merged**